lib/solvers/TraceLabelOverlapAvoidanceSolver/pathUtils.ts

```diff
@@ -0,0 +1,79 @@
+import type { Point } from "graphics-debug"
+import {
+  isVertical,
+  isHorizontal,
+  segmentIntersectsRect,
+} from "../SchematicTraceLinesSolver/SchematicTraceSingleLineSolver2/collisions"
+
+export const countTurns = (points: Point[]): number => {
```

Comment:
split into their own files, it's easier to find via search and better for ai context gathering